tensorflow_addons/layers/crf.py

```diff
@@ -0,0 +1,478 @@
+            "bias_initializer":
+            tf.keras.initializers.serialize(self.bias_initializer),
+            "activation":
+            tf.keras.activations.serialize(self.activation),
+            "kernel_regularizer":
+            tf.keras.regularizers.serialize(self.kernel_regularizer),
+            "chain_regularizer":
+            tf.keras.regularizers.serialize(self.chain_regularizer),
+            "boundary_regularizer":
+            tf.keras.regularizers.serialize(self.boundary_regularizer),
+            "bias_regularizer":
+            tf.keras.regularizers.serialize(self.bias_regularizer),
+            "kernel_constraint":
+            tf.keras.constraints.serialize(self.kernel_constraint),
+            "chain_constraint":
+            tf.keras.constraints.serialize(self.chain_constraint),
+            "boundary_constraint":
+            tf.keras.constraints.serialize(self.boundary_constraint),
+            "bias_constraint":
+            tf.keras.constraints.serialize(self.bias_constraint)
+        }
+        base_config = super(CRF, self).get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+    def compute_output_shape(self, input_shape):
+        output_shape = input_shape[:2]
+        return output_shape
+
+    def compute_mask(self, input_, mask=None):
+        """keep mask shape [batch_size, max_seq_len]"""
+        return mask
+
+    def get_negative_log_likelihood(self, y_true):
+        # TODO(howl-anderson): remove unnecessary typing cast
+        self.potentials = tf.keras.backend.cast(self.potentials, tf.float32)
+        y_true = tf.keras.backend.cast(y_true, tf.int32)
+        self.sequence_length = tf.keras.backend.cast(self.sequence_length,
+                                                     tf.int32)
+        # self.chain_kernel = tf.keras.backend.cast(self.chain_kernel,
+        #                                           tf.float32)
+
+        log_likelihood, _ = crf_log_likelihood(
+            self.potentials, y_true, self.sequence_length, self.chain_kernel)
+
+        return -log_likelihood
+
+    def get_loss(self, y_true, y_pred):
+        # we don't use y_pred, but caller pass it anyway, ignore it
+        return self.get_negative_log_likelihood(y_true)
+
+    def get_accuracy(self, y_true, y_pred):
+        judge = tf.keras.backend.cast(
+            tf.keras.backend.equal(y_pred, y_true), tf.keras.backend.floatx())
+        if self.mask is None:
+            return tf.keras.backend.mean(judge)
+        else:
+            mask = tf.keras.backend.cast(self.mask, tf.keras.backend.floatx())
+            return (tf.keras.backend.sum(judge * mask) /
+                    tf.keras.backend.sum(mask))
+
+    def _dense_layer(self, input_):
+        if self.use_kernel:
+            output = self.activation(
+                tf.keras.backend.dot(input_, self.kernel) + self.bias)
+        else:
+            output = input_
```

Comment:
Any chance of replacing this with a `tf.keras.layers.Dense`?

Reply:
fixed